gobblin-data-management/src/main/java/gobblin/data/management/copy/hive/HiveCopyEntityHelper.java

```diff
@@ -889,8 +802,20 @@ private static void checkPartitionCompatibility(Partition desiredTargetPartition
         referenceAuthority = uri.getAuthority();
       }
 
+      if (!this.dataset.getTableRootPath().isPresent()) {
```

Comment:
One performance concern: if there are a lot of files and they are in the same sub directory, if the first file's ancestorOwnerAndPermission been calculated, all other files should not need to re-calculate the ancestorOwnerAndPermission, right?